scrub-logs.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+
+##===---------- scrub-logs.py - generate README from Wiki sources ---------===##
+#
+#                      The LLVM Compiler Infrastructure
+#
+# This file is distributed under the University of Illinois Open Source
+# License. See LICENSE.TXT for details.
+#
+##===----------------------------------------------------------------------===##
+
+"""Scrub irrelevant details from IWYU/Clang logs.
+
+When Clang changes upstream, we usually look for differences in the AST to
+explain the new behavior. This script makes that easier by scrubbing pointer
+values and path prefixes from ast-dump output, so they can be diffed directly.
+"""
+
+import re
+import sys
+
+
+def strip_path_prefix(line):
+    line = re.sub(r'<.*(llvm[\\/]tools[\\/].*):', r'<\1:', line)
+    return line
+
+
+def strip_addrs(line):
+    line = re.sub('\s(0x)?[0-9A-Fa-f]{6,9}', '', line)
```

Comment:
On macOS I have addresses like 0x7fe1d81b3e10 which aren't caught by your regexp. The tricky part is it will apply to the part of the address leaving it in inconsistent state. And I don't know if it is better to remove addresses entirely or to replace with something generic. I was replacing it with 0xxxxx, for instance.